services/galley/src/Galley/API/Teams.hs

```diff
@@ -196,7 +197,14 @@ uncheckedDeleteTeam zusr zcon tid = do
         (ue, be) <- foldrM (pushConvDeleteEvents now membs) ([],[]) convs
         let e = newEvent TeamDelete tid now
         let r = list1 (userRecipient zusr) (membersToRecipients (Just zusr) membs)
-        pushSome ((newPush1 zusr (TeamEvent e) r & pushConn .~ zcon) : ue)
+        -- To avoid DoS on gundeck, send team deletion events in chunks
+        let chunks = List.chunksOf 128 (toList r)
+        forM_ chunks $ \chunk -> case chunk of
+            [] -> return ()
+            x:xs -> push1 (newPush1 zusr (TeamEvent e) (list1 x xs) & pushConn .~ zcon)
+        -- To avoid DoS on gundeck, send conversation deletion events slowly
+        forM_ ue $ \event ->
```

Comment:
How about sleeping _a bit_ also here (maybe even 10ms would help) before `push1 event` ? These can be a lot of events (basically unbound as mentioned) so it could help too.

Reply:
These can't be unbounded, each `event` here can be as long as the amount of participants of one single conversation. Or am I reading the code wrong?
It's onbounded when sending all of them in a `pushSome ue`, which is what it did before.

Reply:
Right, I just meant the `ue` is a list of infinite size, right? The size is basically nr of conversations where non team members are present, right? So if we have 1k events, we will do 1k requests to gundeck, right? Or am _I_ reading it wrong? :)